extensions/resteasy-jaxb/deployment/src/main/java/io/quarkus/resteasy/jaxb/deployment/ResteasyJaxbProcessor.java

```diff
@@ -1,14 +1,54 @@
 package io.quarkus.resteasy.jaxb.deployment;
 
+import java.lang.annotation.Annotation;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+import org.jboss.resteasy.annotations.providers.jaxb.Wrapped;
+import org.jboss.resteasy.annotations.providers.jaxb.WrappedMap;
+
 import io.quarkus.deployment.Feature;
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 
 public class ResteasyJaxbProcessor {
 
+    private static final List<Class<? extends Annotation>> RESTEASY_JAXB_ANNOTATIONS = Arrays.asList(
+            Wrapped.class,
+            WrappedMap.class);
+
+    @Inject
+    BuildProducer<ReflectiveClassBuildItem> reflectiveClass;
+
+    @BuildStep
+    void processAnnotations(CombinedIndexBuildItem combinedIndexBuildItem) {
+        IndexView index = combinedIndexBuildItem.getIndex();
+
+        // Handle RESTEasy annotations usage.
+        for (Class annotationClazz : RESTEASY_JAXB_ANNOTATIONS) {
+            DotName annotation = DotName.createSimple(annotationClazz.getName());
+
+            if (!index.getAnnotations(annotation).isEmpty()) {
+                addReflectiveClass(true, true, "org.jboss.resteasy.plugins.providers.jaxb.JaxbCollection");
+                addReflectiveClass(true, true, "org.jboss.resteasy.plugins.providers.jaxb.JaxbMap");
```

Comment:
Are the providers automatically added to RESTEasy somehow? If so I'm surprised they are not added for reflection there.

Reply:
Not sure to see what you mean... This solely based on experiment ;-) I do not have specific knowledge of RestEasy internal. Do you mean there's some other provider classes that should be required too ?

Reply:
I'm just a bit surprised that just registering them for reflection can make things work.

I would have expected either it would work as is or adding them for reflection won't be enough.

Reply:
Yes just work as is ;-)